pkg/curatedpackages/packageclient.go

```diff
@@ -46,3 +59,74 @@ func convertBundleVersionToPackageVersion(bundleVersions []packagesv1.SourceVers
 	}
 	return versions
 }
+
+func (pc *PackageClient) GeneratePackages() ([]packagesv1.Package, error) {
+	packageMap := pc.getPackageNameToPackage()
+	var packages []packagesv1.Package
+	for _, p := range pc.packages {
+		bundlePackage := packageMap[strings.ToLower(p)]
+		if bundlePackage.Name == "" {
+			fmt.Println(fmt.Errorf("unknown package %s", p).Error())
+			continue
+		}
+		name := CustomName + strings.ToLower(bundlePackage.Name)
+		packages = append(packages, convertBundlePackageToPackage(bundlePackage, name, pc.bundle.APIVersion))
+	}
+	return packages, nil
+}
+
+func (pc *PackageClient) WritePackagesToFile(packages []packagesv1.Package, d string) error {
+	directory := filepath.Join(d, packageLocation)
+	if !validations.FileExists(directory) {
+		if err := os.Mkdir(directory, dirPermission); err != nil {
```

Comment:
```suggestion
		if err := os.MkdirAll(directory, dirPermission); err != nil {
```
[`os.MkdirAll`](https://pkg.go.dev/os#MkdirAll)

Reply:
Why is this necessary?

Reply:
[Mkdir](https://pkg.go.dev/os#Mkdir) versus [MkdirAll](https://pkg.go.dev/os#MkdirAll)

TL;DR: MkdirAll will create any missing directories along the way; so it's just safer. The only reason _not_ to use it, is if creating more than a single level of the directory hierarchy would be an error.